frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx

```diff
@@ -1418,12 +1426,22 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
             },
         ],
         replayFilters: [
-            (s) => [s.webAnalyticsFilters, s.dateFilter, s.shouldFilterTestAccounts],
+            (s) => [s.webAnalyticsFilters, s.dateFilter, s.shouldFilterTestAccounts, s.conversionGoal],
             (
                 webAnalyticsFilters: WebAnalyticsPropertyFilters,
                 dateFilter,
-                shouldFilterTestAccounts
+                shouldFilterTestAccounts,
+                conversionGoal
             ): RecordingUniversalFilters => {
+                const filters: UniversalFiltersGroupValue[] = [...webAnalyticsFilters]
+                if (conversionGoal) {
+                    const actionId = (conversionGoal as ActionConversionGoal).actionId
+                    const customEventName = (conversionGoal as CustomEventConversionGoal).customEventName
+                    const id = actionId || customEventName
+
+                    filters.push({ id, name: String(id), type: customEventName ? 'events' : 'actions' })
+                }
```

Comment:
Is there an easier way to do this, maybe?

Reply:
Yes definitely, you can do 

`if "actionId" in conversionGoal` or `if "customEventName" in conversionGoal` and TypeScript will do [narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) to give you the correct types inside that branch of the if statement